### DISCUSSION TITLE
arduino-esp32 example of eth.h with w5500

### DISCUSSION THREAD

**Discussion by Rgrndaal** [espressif/arduino-esp32#6548] [Q&A]
I'm trying to figure out how spi-ethernet (ETH.H) with a w5500 can be used with esp32-arduino 2.0.3rc1. With the arduino ethernet.h this works like a charm but WiFiServer or WiFiClient cannot be used, and one would need to implement servers and clients twice for both wifi and ethernet. 

What's the Arduino approach here? Is esp-idf wrapped somewhere or does one simply use the functions from the headers directly:
#include "esp_netif.h"
#include "esp_eth.h"
#include "esp_event.h"
#include "esp_log.h"
#include "driver/gpio.h"
#include "sdkconfig.h"
#include "driver/spi_master.h"

DM9051 is initialised through ETH.begin()

**Comment by razvanphp** [comment]
This feature was added in `v3.x`

https://github.com/espressif/arduino-esp32/pull/7163#issuecomment-1771166635

https://github.com/espressif/arduino-esp32/pull/8712
